Automattic/jetpack
# VideoPress: Add chapters generation from description to dashboard

Fixes #27750

## Proposed changes:

* Extracts the live parsing from the block editor to a hook, reused in the dashboard video details edit page

*Known issue*: It does not yet check for the existence of non-auto-generated chapters. Reason: https://github.com/Automattic/jetpack/issues/27492

### Other information:


## Jetpack product discussion


N/A

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:






* Go to the block editor
* Add a VideoPress block with a video
* Save the post and close the page (to avoid any conflict with auto-save in the block editor)
* Go to the VideoPress dashboard and find the video used in the block editor
* In the Details page, add chapters in the description
* Check that there is a helper message with a modal about the use of chapters, same as in the block editor
* Check that after adding an incomplete chapters list and waiting for 3 seconds, the message changes to a notice, same as in the block editor
* After completing the chapters list, save the changes
* You can check the console for a `debug` message: `videopress:use-edit-details Autogenerated track: [...]`
* Reopen the block editor with the video block
* Check that the new chapters are reflected in the video
* Check that the block editor behavior of changing chapters still works as expected
* Try it on a video that has no chapters, one that has auto-generated chapters (by the block) and one with uploaded tracks. The uploaded tracks must not be overwritten: `videopress:use-edit-details Chapter people-pouring-a-warm-drink-around-a-campfire-32_chapters_en-14.vtt detected. Overwritable: no`

https://user-images.githubusercontent.com/8486249/221228891-edcdf4ef-4f23-4876-9bbc-a839195bcb4a.mp4
